docs/adr/0005-feature-flags.md

```diff
@@ -0,0 +1,149 @@
+1. section (optional): we present the name of the section of the app the feature is gating (free/premium). 
+2. purpose: indicates what the feature does (can be tied to an epic name)
+3. service: where in the stack the feature is located (subsection of an epic)
+
+An **example** of such names would be:
+```
+premium_subplat_email
+```
+
+In this example, the feature is gating functionality in the "settings" section. It is a part of the "premium" feature. The main area of concern is the "email"
+service.
+
+### Retiring Flags / Zombies
+One of the worst problems when an org adopts feature flags is that the number of flags within a product can grow to become overwhelming. No one is entirely sure who’s responsible for which flags or which flags are no longer being used.
+
+We need to have a process in place where flags are retired after they serve their purpose. This can be done as a part of our agile process (grooming /planning) where we always have a task in the working backlog to clean up the feature flags that were created for the current feature. It's important to not mark this task as tech debt though, since tech debts can have a nasty tendency of being perpetually deprioritized.
+
+For more drastic measure, there are teams that opt to place a limit on the number of active flags they have under management. This incentivizes the removal of old flags to make room for a new flag.
+
+It's much easier to spot the zombie flags if our code is being maintained properly after feature complete. The if/elses that usually associate with feature flags should be deleted upon completition of a feature, and with a simple search, we should be able to quickly spot stale feature flags
+
+### Ownership
+Looking ahead, there may come a time when we need a system that enables the ability to assign ownership of a flag
+
+Feature flags offer a powerful way to directly modify the behavior of production systems, acting as a kill switch. The configuration of flags should undergo the same level of change control as the deployment of production code. Associating explicit ownership with a flag allows for the application of change control policies. For instance, individuals can be restricted from modifying flags they do not own, or approval can be required before making feature-flag configuration changes in the production environment.
+
+### Flag Tagging
+Attaching information like expiration dates and ownership to a flag are specific examples of a more general capability: the ability to tag flags.
+
+Tagging refers to the ability to attach arbitrary key:value pairs to a feature flag. This concept is employed in infrastructure systems like Kubernetes and AWS. It provides a highly adaptable method for users to associate semantic metadata with entities within the system. Tags can be used to track various aspects of a flag, such as its creation date, expiration date, life-cycle status, flag type, ownership, authorized modifier etc.
+
+## Considered Options
```

Comment:
Another option could be having a JSON file with a data structure that we define. When it comes to infrastructure that option would be similar to an `.env`, but it could give us still the ability to add more context like tagging, sub-feature flags and  dependencies on other features. Something like:
```
{
  "flags": [
    {
      "name": "Party mode",
      "key": "party-mode",
      "party-mode": {
        "dev": true,
        "prod": false
      },
      "dependencies": ["confetti"]
    },
    {
      "name": "Confetti",
      "key": "confetti",
      "party-mode": {
        "dev": true,
        "prod": false
      },
      "dependencies": []
    }
  ]
}
```

Reply:
I agree, this is also a good DIY option. A lot more extensible. I'll make a separate option for this

Reply:
Added `JSON` as an option. Thanks @flozia